activerecord/lib/active_record/fixtures.rb

```diff
@@ -530,38 +479,129 @@ def default_fixture_model(fs_name, config)
         end
     end
 
-    def self.create_fixtures(fixtures_directory, fixture_set_names, class_names = {}, config = ActiveRecord::Base)
-      fixture_set_names = Array(fixture_set_names).map(&:to_s)
-      class_names = ClassCache.new class_names, config
+    class << self
+      def default_fixture_model_name(fixture_set_name, config = ActiveRecord::Base) # :nodoc:
+        config.pluralize_table_names ?
+          fixture_set_name.singularize.camelize :
+          fixture_set_name.camelize
+      end
 
-      # FIXME: Apparently JK uses this.
-      connection = block_given? ? yield : ActiveRecord::Base.connection
+      def default_fixture_table_name(fixture_set_name, config = ActiveRecord::Base) # :nodoc:
+        "#{ config.table_name_prefix }"\
+        "#{ fixture_set_name.tr('/', '_') }"\
+        "#{ config.table_name_suffix }".to_sym
+      end
 
-      files_to_read = fixture_set_names.reject { |fs_name|
-        fixture_is_cached?(connection, fs_name)
-      }
+      def reset_cache
+        @@all_cached_fixtures.clear
+      end
 
-      unless files_to_read.empty?
-        fixtures_map = {}
+      def cache_for_connection(connection)
+        @@all_cached_fixtures[connection]
+      end
+
+      def fixture_is_cached?(connection, table_name)
+        cache_for_connection(connection)[table_name]
+      end
+
+      def cached_fixtures(connection, keys_to_fetch = nil)
+        if keys_to_fetch
+          cache_for_connection(connection).values_at(*keys_to_fetch)
+        else
+          cache_for_connection(connection).values
+        end
+      end
+
+      def cache_fixtures(connection, fixtures_map)
+        cache_for_connection(connection).update(fixtures_map)
+      end
+
+      def instantiate_fixtures(object, fixture_set, load_instances = true)
+        return unless load_instances
+        fixture_set.each do |fixture_name, fixture|
+          begin
+            object.instance_variable_set "@#{fixture_name}", fixture.find
+          rescue FixtureClassNotFound
+            nil
+          end
+        end
+      end
+
+      def instantiate_all_loaded_fixtures(object, load_instances = true)
+        all_loaded_fixtures.each_value do |fixture_set|
+          instantiate_fixtures(object, fixture_set, load_instances)
+        end
+      end
+
+      def create_fixtures(fixtures_directory, fixture_set_names, class_names = {}, config = ActiveRecord::Base)
+        fixture_set_names = Array(fixture_set_names).map(&:to_s)
+        class_names = ClassCache.new class_names, config
+
+        # FIXME: Apparently JK uses this.
```

Comment:
This is not in the scope of this PR, but maybe we can get rid of this comment, it has already been here for many years...

Reply:
Yeah, I was meaning to comment on this as well. We will address it after some feedback from the core team. 👍

Reply:
JK also became JD in the intervening years 😄 — Now I'm curious @jeremy, are you still using it?

Reply:
Let's leave it. Normally I'd say a test is sufficient but I know this comment has stopped me from messing with this line before and I'd rather not break Basecamp's fixtures. Also I get a good laugh from it whenever I see it. 😉

Reply:
Asking here out of curiosity and behaviour I discovered today around this method

1. What is JK or JD here?
2. `connection = block_given? ? yield : ActiveRecord::Base.connection` accepts a connection as a block to `create_fixtures` which appratenly fixed an issue I was having for my project's test suite where I was able to create fixtures for different shards in my `test_helper.rb`, purely because of this line. Rougly I ended up doing something like this,
```ruby
# worked
ActiveRecord::FixtureSet.create_fixtures('test/fixtures', 'posts') { ActiveRecord::Base.connecting_to(shard: :a) }
```
For some reason,
```ruby
# This did not work
ActiveRecord::Base.connected_to(shard: :a) { ActiveRecord::FixtureSet.create_fixtures('test/fixtures', 'posts') }
```
@eileencodes any thoughts on the behaviour?